### PR TITLE
Adding nn.SelectCriterion

### PR DESCRIPTION
**Pull request by lopezpaz**
Analogous to nn.SelectTable, but for criterions.

### CODE REVIEW

**Comment by soumith** [comment]
The builds fail. it is because you've not added SelectCriterion to init.lua
https://travis-ci.org/torch/nn/builds/114287408

**Comment by lopezpaz** [comment]
Added.

File: SelectCriterion.lua [hunk 1]
```diff
@@ -0,0 +1,51 @@
+local SelectCriterion, parent = torch.class('nn.SelectCriterion', 'nn.Module')
```
Comment (soumith) [inline SelectCriterion.lua]: derive from nn.Criterion

**Comment by ebetica** [comment]
FYI the build is failing because of a bug with the tester. This has been fixed so triggering a rebuild will pass it.

**Comment by soumith** [comment]
this PR is fairly wrong, so i'll just close it for now.

**Comment by nicholas-leonard** [comment]
You can use dpnn's [ModuleCriterion](https://github.com/Element-Research/dpnn/blob/master/README.md#nn.ModuleCriterion) to do this kind of thing for any module, not just SelectTable. Also works on either inputs or targets (or both). Would you like a PR?

**Comment by fmassa** [comment]
I think this ModuleCriterion would be a nice addition !

**Pull request closed without merging**